src/efi.rs

```diff
@@ -381,10 +382,42 @@ impl Component for Efi {
     }
 
     fn generate_update_metadata(&self, sysroot_path: &str) -> Result<ContentMetadata> {
-        let ostreebootdir = Path::new(sysroot_path).join(ostreeutil::BOOT_PREFIX);
         let dest_efidir = component_updatedir(sysroot_path, self);
 
-        if ostreebootdir.exists() {
+        // New EFI dir /usr/lib/efi
+        let efi_lib = Path::new(sysroot_path).join("usr/lib/efi");
+        // Legacy ostree-boot dir
+        let ostreebootdir = Path::new(sysroot_path).join(ostreeutil::BOOT_PREFIX);
+
+        let meta = if efi_lib.exists() {
+            // Get EFI paths like /usr/lib/efi/(shim|grub2)/<version>/EFI
+            let efi_paths = find_all_efi_dirs(&efi_lib)
+                .with_context(|| format!("Failed to find EFI path under {}", efi_lib.display()))?;
+
+            // Get all files under EFI paths
+            let all_files: Result<Vec<String>> =
+                efi_paths.iter().try_fold(Vec::new(), |mut acc, efi| {
+                    let efidir = openat::Dir::open(efi)
+                        .with_context(|| format!("Opening {}", efi.display()))?;
+
+                    // Get all files path
+                    let mut files = crate::util::filenames(&efidir)?
+                        .into_iter()
+                        .map(|mut f| {
+                            f.insert_str(0, efi.to_str().unwrap());
+                            f
+                        })
+                        .collect::<Vec<String>>();
+
+                    Command::new("mv")
```

Comment:
I think we have an opportunity here to make `usr/lib/efi` the standard path actually if it exists, maybe we could make things as simple as the equivalent of `ln -sr /usr/lib/efi /usr/lib/bootupd/updates/EFI`?

EDIT: Ah I see it's not that simple based on `find_all_efi_dirs`.

Hmmm...but actually I like the idea of that layout, what if we tried to adopt *that* as the standard and migrate our current EFI layout to it? (It'd break updates for older bootupd though...without having dual layouts for a while, ug)